File: lib-database/src/main/java/com/ustadmobile/core/db/dao/ContentCategoryDao.java
package com.ustadmobile.core.db.dao;

import com.ustadmobile.lib.database.annotation.UmDao;
import com.ustadmobile.lib.database.annotation.UmQuery;
import com.ustadmobile.lib.database.annotation.UmRepository;
import com.ustadmobile.lib.database.annotation.UmUpdate;
import com.ustadmobile.lib.db.entities.ContentCategory;
import com.ustadmobile.lib.db.sync.dao.SyncableDao;

import java.util.List;

@UmDao(selectPermissionCondition = "(:accountPersonUid = :accountPersonUid)")
@UmRepository
public abstract class ContentCategoryDao implements SyncableDao<ContentCategory, ContentCategoryDao>  {

    @UmQuery("SELECT * FROM ContentCategory WHERE " +
            "ctnCatContentCategorySchemaUid = :schemaId AND name = :name")
    public abstract ContentCategory findCategoryBySchemaIdAndName(long schemaId, String name);

    @UmUpdate
    public abstract void update(ContentCategory entity);

    @UmQuery("SELECT ContentCategory.* FROM ContentCategory")
    public abstract List<ContentCategory> getPublicContentCategories();
}

File: lib-database/src/commonMain/kotlin/com/ustadmobile/core/db/dao/ContentCategoryDao.kt
package com.ustadmobile.core.db.dao

import com.ustadmobile.door.annotation.DoorDao
import androidx.room.Query
import com.ustadmobile.door.annotation.*
import com.ustadmobile.lib.db.entities.ContentCategory

@Repository
@DoorDao
expect abstract class ContentCategoryDao : BaseDao<ContentCategory> {

    @Query("SELECT ContentCategory.* FROM ContentCategory")
    abstract fun publicContentCategories(): List<ContentCategory>

    @Query("SELECT * FROM ContentCategory WHERE " + "ctnCatContentCategorySchemaUid = :schemaId AND name = :name")
    abstract fun findCategoryBySchemaIdAndName(schemaId: Long, name: String): ContentCategory?
}
